### ISSUE TITLE
map.jinja is overwriting not merging maps

### ISSUE THREAD

**Issue by noelmcloughlin** [saltstack-formulas/docker-formula#219]
The following error (on macos) shows that `defaults` and `osfamilymap` merge is broken.
```
local:
    Data failed to compile:
----------
    Rendering SLS 'base:docker' failed: Jinja variable 'dict object' 
has no attribute 'allow_updates'
```

**Comment by myii** [comment]
@noelmcloughlin The issue is the multiple use of `.update` for dicts in `map.jinja`, which overwrite keys instead of merging them.  Migrating to the new style `map.jinja` will resolve this.

**Comment by noelmcloughlin** [comment]
Yes, that explanation sounds right. This formula needs to be migrated!!

**Comment by noelmcloughlin** [comment]
Fixed in merged #256 (hopefully ;-)
